buildcontext/git.go

```diff
@@ -153,9 +153,10 @@ func (gr *gitResolver) resolveGitProject(ctx context.Context, gwClient gwclient.
 			llb.WithCustomNamef("[internal] GIT CLONE %s", stringutil.ScrubCredentials(gitURL)),
 			llb.KeepGitDir(),
 		}
-		if keyScan != "" {
-			gitOpts = append(gitOpts, llb.KnownSSHHosts(keyScan))
+		if len(keyScans) > 0 {
+			gitOpts = append(gitOpts, llb.KnownSSHHosts(strings.Join(keyScans, "\n")))
```

Comment:
Maybe a naive question, but I didn't get the impression that llb.KnownSSHHosts receives multiple keys separated by new lines, is that documented somewhere else? Or were you assuming it accepts any content compatible with `~/.ssh/authorized_keys?

Reply:
it passes it along to buildkit which writes it out to a temp file which is then passed to `git` when cloning: https://github.com/moby/buildkit/blob/a0afb6902ad2014f702f196ae2e597c1768a763b/source/git/gitsource.go#L283

perhaps the original author of the known_hosts implementation in buildkit could have done a better job documenting that: https://github.com/moby/buildkit/commit/f2c8eb177c83f1381e69966bdd0312dd9d86e1ff :laughing:

Reply:
LOLL -- thanks good to know :)